src/common/PatchDB.cpp

```diff
@@ -839,7 +839,6 @@ CREATE TABLE IF NOT EXISTS Favorites (
         std::ostringstream searchName;
         searchName << p.name << " ";
 
-        if (storage)
```

Comment:
The static analyzer is sort of correct here, that storage is used above, but it is only used in an exception path. The correct fix I think is not to remove this guard, but to guard the report errors above. I'll make that change on the branch.

Reply:
@baconpaul IIRC there are many places where it's used unguarded and I couldn't think why it should only not be null only in exception paths - which was why I assumed it probably never is null in practice!

Reply:
In this function it's only used prior in the exception path. so i flipped it to check the exception paths

Those exceptions are pretty rare so I think if storage was null those cases would crash.

I sort of wish there was an easy non-nullable pointer type. I guess that's a & but they are clumsy to mutate.

Reply:
The constructor on L413 would crash were it ever passed a nullptr for storage.
Reworking using a ref for storage seems to build fine for me.

Reply:
Yeah i think you are correct.

Since synth : storage : ui are all 1:1 in surge right now, we really should have set everything up as a & or const&. That's more what we are doing in short circuit. Many of the 2004 bones are still on display in the raw pointer idiom here.